### PR TITLE
removed zplots context and changed to a singleton

### PR DESCRIPTION
**Pull request by zack-erisyon**

### CODE REVIEW

File: plaster/tools/zplots/zplots.py [hunk 1]
```diff
@@ -1468,7 +1470,9 @@ def setup():
     np.set_printoptions(precision=1)
     output_notebook()
     notebook_full_width()
-    return ZPlots()
+    ZPlots.zplot_singleton is None
```
Comment (tblom) [inline plaster/tools/zplots/zplots.py:1473]: I assume you want an "if" here, otherwise we create a new one each time.  :)

**Review by tblom — CHANGES_REQUESTED** [review]

File: plaster/tools/zplots/zplots.py (hunk 1 above)
Comment (tblom) [reply plaster/tools/zplots/zplots.py:1473]: Seems like some kind of python linter would catch this?

Comment (zack-erisyon) [reply plaster/tools/zplots/zplots.py]: Fixed. Was meant ot be an assert

Comment (tblom) [reply plaster/tools/zplots/zplots.py]: Right.  My initial comment was I thought you were missing an assert, but then I was confused by the class returning the singleton, as if this was an instantiation and not a class initialization.  What does it mean for class initialization to return something?  Doesn't this run at module load?

Comment (zack-erisyon) [reply plaster/tools/zplots/zplots.py]: This isn't a class init. This is a free function. It just happens to jam the singleton back in to the class. It it a little weird, I admit

Comment (tblom) [reply plaster/tools/zplots/zplots.py]: hah, my confusion was due to diff formatting.  Looked like it was inside "class Zplots:"  defn, I overlooked line numbers.  :)

**Pull request merged**